components/dashboard/src/components/DropDown2.tsx

```diff
@@ -0,0 +1,201 @@
+            setTimeout(() => {
+                // only close if the focussed element is not child
+                if (!nodeRef?.current?.contains(window.document.activeElement)) {
+                    setShowDropDown(false);
+                }
+            }, 100);
+        },
+        [setShowDropDown],
+    );
+
+    return (
+        <div
+            onKeyDown={onKeyDown}
+            onBlur={handleBlur}
+            ref={nodeRef}
+            tabIndex={0}
+            className={"relative flex flex-col rounded-lg  focus:outline-none focus:ring-2 focus:ring-blue-300"}
+        >
+            <div
+                className={
+                    "h-16 bg-gray-100 dark:bg-gray-800 flex items-center px-2 " +
+                    (showDropDown
+                        ? "rounded-t-lg"
+                        : "rounded-lg hover:bg-gray-200 dark:hover:bg-gray-700 cursor-pointer")
+                }
+                onClick={toggleDropDown}
+            >
+                {props.children}
+                <div className="flex-grow" />
+                <div className="mr-2">
+                    <Arrow direction={showDropDown ? "up" : "down"} />
+                </div>
+            </div>
+            {showDropDown && (
+                <>
+                    <div className="absolute w-full top-12 bg-gray-100 dark:bg-gray-800 max-h-72 overflow-auto rounded-b-lg mt-3 z-50 p-2">
+                        {!props.disableSearch && (
+                            <div className="h-12">
+                                <input
+                                    type="text"
+                                    autoFocus
+                                    className={"w-full focus rounded-lg"}
+                                    placeholder={props.searchPlaceholder}
+                                    value={search}
+                                    onChange={(e) => setSearch(e.target.value)}
+                                />
+                            </div>
+                        )}
+                        <ul>
+                            {filteredOptions.length > 0 ? (
+                                filteredOptions.map((element) => {
+                                    let selectionClasses = `dark:bg-gray-800 cursor-pointer`;
+                                    if (element.id === selectedElementTemp) {
+                                        selectionClasses = `bg-gray-200 dark:bg-gray-700 cursor-pointer  focus:outline-none focus:ring-0`;
+                                    }
+                                    if (!element.isSelectable) {
+                                        selectionClasses = ``;
+                                    }
+                                    return (
+                                        <li
+                                            key={element.id}
+                                            id={element.id}
+                                            tabIndex={0}
```

Comment:
I wonder if we want to avoid having these option elements be tab indexed. As it is, once the container is open, tabbing doesn't proceed to the next "input" in the view, i.e. the next dropdown in this case, but shifts the selected option to the next one. A normal select in the browser will only use arrow keys for navigating the selected option, while tab will let you move on to the next "input".